core/src/main/java/io/questdb/griffin/engine/ops/Operation.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.ops;
+
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.sql.OperationFuture;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.mp.SCSequence;
+import io.questdb.std.QuietCloseable;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Operation represents immutable user request to execute a DDL against the database. It aims to
+ * separate compilation and execution phases of SQL query processing. Compilation phase is
+ * text parsing with parse results stored in the Operation object. Execution phase is a process
+ * of applying the request to the current context. Operation can be executed multiple times.
+ * <p>
+ * When operation is no longer needed, it must be closed to release resources.
+ */
+public interface Operation extends QuietCloseable {
```

Comment:
nit: it should be called `DdlOperation`? Since there already is `InsertOperation` and `AlterOperation`. 
Have a look at CompiledQuery:
```java
@Transient
    AlterOperation getAlterOperation();
    Operation getOperation();
    InsertOperation getInsertOperation();
```

Reply:
those two will eventually roll under the "Operation", it is an incremental change :(